src/files/FilesPage.js

```diff
@@ -57,14 +54,12 @@ const FilesPage = ({
   }, [files, pinningServices, doFetchRemotePins])
 
   const onDownload = async (files) => {
-    if (downloadProgress !== null) {
-      return downloadAbort()
-    }
+    const { url, filename } = await doFilesDownloadLink(files)
 
-    const updater = (v) => setDownloadProgress(v)
-    const { url, filename, method } = await doFilesDownloadLink(files)
-    const { abort } = await downloadFile(url, filename, updater, method)
-    setDownloadAbort(() => abort)
+    const link = document.createElement('a')
+    link.download = filename // TODO: filename isn't working. Keep getting 'get.tar.gz' - is it being overwritten?
```

Comment:
Oh wow, indeed, `download` attribute only works on links within the same Origin or.. `blob:` ([MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#attr-download)), given that users load webui from different port or domain than the RPC API, we can't use it here :crying_cat_face: 

We can link to standalone files directly, using gateway link, 
but for  directories that need to be returned as `tar.gz` we have no good option for setting filename on ipfs-webui side.

Unless someone comes up with a creative hack, I am afraid we need to park this PR until we get something like  [`?format=tar.gz`](https://github.com/ipfs/go-ipfs/issues/8234) on Gateways, or  `/api/v0/get` with support for  implicit or explicit `filename` parameter that modifies response and adds two HTTP headers:
```
Content-Type: application/octet-stream
Content-Disposition: attachment; filename=cid.tar.gz
```

I'd  rather wait for clean `?format=tar.gz`  (hoping to start working on it this quarter) as we want to remove `/api/v0` from gateways.

Reply:
@lidel please see #1945

Reply:
For anyone reading this in the future, this got resolved in https://github.com/ipfs/kubo/pull/9029, we can switch to that.